pyomo/common/numeric_types.py

```diff
@@ -64,34 +69,53 @@
 #: like numpy.
 #:
 #: :data:`native_types` = :data:`native_numeric_types <pyomo.core.expr.numvalue.native_numeric_types>` + { str }
-native_types = set([bool, str, type(None), slice, bytes])
+native_types = {bool, str, type(None), slice, bytes}
 native_types.update(native_numeric_types)
 native_types.update(native_integer_types)
-native_types.update(_native_boolean_types)
+native_types.update(native_complex_types)
 native_types.update(native_logical_types)
+native_types.update(_native_boolean_types)
 
 nonpyomo_leaf_types.update(native_types)
 
 
-def RegisterNumericType(new_type):
-    """
-    A utility function for updating the set of types that are
-    recognized to handle numeric values.
+def RegisterNumericType(new_type: type):
+    """Register the specified type as a "numeric type".
+
+    A utility function for registering new types as "native numeric
+    types" that can be leaf nodes in Pyomo numeric expressions.  The
+    type should be compatible with :py:class:`float` (that is, store a
+    scalar and be castable to a Python float).
+
+    Parameters
+    ----------
+    new_type: type
+        The new numeric type (e.g, numpy.float64)
 
-    The argument should be a class (e.g, numpy.float64).
     """
     native_numeric_types.add(new_type)
     native_types.add(new_type)
     nonpyomo_leaf_types.add(new_type)
 
 
-def RegisterIntegerType(new_type):
-    """
-    A utility function for updating the set of types that are
-    recognized to handle integer values. This also registers the type
-    as numeric but does not register it as boolean.
+def RegisterIntegerType(new_type: type):
+    """Register the specified type as an "integer type".
+
+    A utility function for registering new types as "native integer
+    types".  Integer types can be leaf nodes in Pyomo numeric
+    expressions.  The type should be compatible with :py:class:`float`
+    (that is, store a scalar and be castable to a Python float).
```

Comment:
Believe it or not, it is not a typo.  The assumption in the Numeric expression system is that the value of any expression should be float-like.  

This set is actually not used *anywhere* in Pyomo.  I think it was added in the dawn of time on speculation that it might be useful.  Maybe the right thing is to deprecate it entirely?

Reply:
I would be in favor of deprecating it before someone starts using it and we get really confused. :P